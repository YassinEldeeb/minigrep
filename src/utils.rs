use colored::Colorize;
use std::fmt::Display;

pub fn print_error<T, U>(error_type: T, error_msg: U) -> String
where
    T: Display + Colorize,
    U: Display,
{
    let error = format!("{}{} {}", error_type.red(), ":".bold(), error_msg);
    println!("{}", error);

    error
}

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_debug_snapshot;

    #[test]
    fn format_error_properly() {
        let err_msg = print_error("Memory Leak", "something went wrong");

        assert_debug_snapshot!(err_msg);
    }
}
