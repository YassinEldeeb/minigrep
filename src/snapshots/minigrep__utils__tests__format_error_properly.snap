---
source: src/utils.rs
assertion_line: 24
expression: err_msg

---
"\u{1b}[31mMemory Leak\u{1b}[0m\u{1b}[1m:\u{1b}[0m something went wrong"
